packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -421,9 +463,9 @@ function reconcile_tracked_array(array, state, anchor, render_fn, flags, keys) {
 		});
 	}
 
-	// TODO: would be good to avoid this closure in the case where we have no
-	// transitions at all. It would make it far more JIT friendly in the hot cases.
-	pause_effects(to_destroy, () => {
+	var controlled_anchor = is_controlled && b_items.length === 0 ? anchor : null;
```

Comment:
This line had the bug before. We now use `b_items.length` rather than `b`, which was a copy bug from the indexed reconcile logic.